File: sample_lock/src/main/java/framework/base/BaseActivity.java
package framework.base;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.lock.R;

/**
 * by y on 2017/2/15
 */

public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        initById();
        initCreate(savedInstanceState);
    }

    protected void replaceFragment(Fragment fragment) {
        getSupportFragmentManager().beginTransaction().replace(R.id.fragment, fragment).commit();
    }

    protected <T extends View> T getView(int id) {
        //noinspection unchecked
        return (T) findViewById(id);
    }


    protected abstract void initCreate(Bundle savedInstanceState);

    protected abstract void initById();

    protected abstract int getLayoutId();

}

File: MVPLifecycle/app/src/main/java/com/android/mvp/lifecycle/base/BaseActivity.kt
package com.android.mvp.lifecycle.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * @author y
 * @create 2019/3/31
 */
abstract class BaseActivity<P : BaseIPresenter> : AppCompatActivity() {

    protected lateinit var presenter: P

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        presenter = initPresenter()
        lifecycle.addObserver(presenter)
    }

    protected abstract fun initPresenter(): P

    override fun onDestroy() {
        super.onDestroy()
        presenter.onDestroy()
    }
}
